From 87e2843ca6aec3f1bda819ec554c389cd897d717 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:43:11 -0400
Subject: [infiniband] cxgb4: Add missing debug stats

Message-id: <1410475447-94392-53-git-send-email-svemuri@redhat.com>
Patchwork-id: 93815
O-Subject: [RHEL7.1 PATCH BZ 1124947 052/108] RDMA/cxgb4: Add missing debug stats
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Signed-off-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 98a3e879907644c0b7e2f16436eb5cf24b9cd61f)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/cxgb4/mem.c b/drivers/infiniband/hw/cxgb4/mem.c
index f9ca072..ec7a298 100644
--- a/drivers/infiniband/hw/cxgb4/mem.c
+++ b/drivers/infiniband/hw/cxgb4/mem.c
@@ -259,8 +259,12 @@ static int write_tpt_entry(struct c4iw_rdev *rdev, u32 reset_tpt_entry,
 
  if ((!reset_tpt_entry) && (*stag == T4_STAG_UNSET)) {
   stag_idx = c4iw_get_resource(&rdev->resource.tpt_table);
-  if (!stag_idx)
+  if (!stag_idx) {
+   mutex_lock(&rdev->stats.lock);
+   rdev->stats.stag.fail++;
+   mutex_unlock(&rdev->stats.lock);
    return -ENOMEM;
+  }
   mutex_lock(&rdev->stats.lock);
   rdev->stats.stag.cur += 32;
   if (rdev->stats.stag.cur > rdev->stats.stag.max)
diff --git a/drivers/infiniband/hw/cxgb4/resource.c b/drivers/infiniband/hw/cxgb4/resource.c
index 94b5fd9..67df71a 100644
--- a/drivers/infiniband/hw/cxgb4/resource.c
+++ b/drivers/infiniband/hw/cxgb4/resource.c
@@ -179,8 +179,12 @@ u32 c4iw_get_qpid(struct c4iw_rdev *rdev, struct c4iw_dev_ucontext *uctx)
   kfree(entry);
  } else {
   qid = c4iw_get_resource(&rdev->resource.qid_table);
-  if (!qid)
+  if (!qid) {
+   mutex_lock(&rdev->stats.lock);
+   rdev->stats.qid.fail++;
+   mutex_unlock(&rdev->stats.lock);
    goto out;
+  }
   mutex_lock(&rdev->stats.lock);
   rdev->stats.qid.cur += rdev->qpmask + 1;
   mutex_unlock(&rdev->stats.lock);
-- 
1.7.1